doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md

```diff
@@ -1772,10 +1779,69 @@ In order to save hard disk space and avoid loss of the audit log records during
+
+### 13.2 Backup archive encryption and verification
+
+Starting from version 7.0 security server backups are signed and optionally encrypted. The GNU Privacy Guard \[[GnuPG](#Ref_GnuPG)\] 
+is used for encryption and signing. GPG keypair is generated during security server initialisation. In addition to generated
+key additional public keys can be used to encrypt backups. 
+
+### 13.3 Backup encryption configuration
+
+Back up encryption is initially turned off. To turn encryption on configuration must be overridden in the file `/etc/xroad/conf.d/local.ini`, in the `[proxy]` section.
+
+    [proxy]
+    backup-encrypted=false
+    backup-public-key-path=/etc/xroad/backupkeys
 
+Backups are always signed. To turn on backup encryption add or change the `backup-encrypted` key to true. If needed also 
+change the folder where additional backup encryption keys are stored. Default is `/etc/xroad/backupkeys`.
 
-### 13.1 Back up and Restore in the User Interface
+Backup encryption keys folder must contain only GPG public key files. In addition to security servers public key backup
+archives are encrypted with all the keys in this folder. It is recommended to use at least one additional public key,
+otherwise the backups will be unusable in case security servers private key is lost. It is up to security servers administrator
+to check that private keys used are sufficiently strong, there are no automatic checks. Additional keys for backup 
+encryption should be generated and stored outside security server in a secure environment.
+
+### 13.4 Verifying backup archive consistency
+
+Security server verifies consistency of backup archives automatically. It is also possible to check archives externally.
+For checking consistency security servers public key is needed. When backups are encrypted then a private key for decrypting
+archive is also needed (GPG uses "sign then encrypt" scheme, so it is not possible to verify encrypted archives without
+decrypting them).
+
+Automatic backup verification is only possible when original security server keypair is available. Should keypair on
+the security server be lost for whatever reason, automatic verification is no longer possible. It is recommended to
+export backup encryption public key and import it into separate secure environment. If backups are encrypted security
+server public key should be imported to keyrings holding additional encryption keys, so that backups can be decrypted
+and verified in these separate environemnts.
+
+To export security servers backup encryption public key use the following command
+
+    gpg --homedir /etc/xroad/gpghome --armor --output server-public-key.gpg --export AA/GOV/TS1OWNER/TS1
+
+where `AA/GOV/TS1OWNER/TS1` is the security server id. 
+
+Resulting file (server-public-key.gpg) should then be exported from security server and imported to GPG keystore used
+for backup archive consistency checking.
```

Comment:
Security server verifies consistency of backup archives automatically. Also, it is possible to verify the consistency of the archives externally. For verifying the consistency externally, security server's public key is needed. When backups are encrypted, then a private key for decrypting archive is also needed. GPG uses "sign then encrypt" scheme, so it is not possible to verify encrypted archives without decrypting them.

Automatic backup verification is only possible when original security server keypair is available. Should keypair on the security server be lost for whatever reason, automatic verification is no longer possible. Therefore, it is recommended to export backup encryption public key and import it into separate secure environment. If backups are encrypted, security server public key should be imported to keyrings holding additional encryption keys, so that backups can be decrypted and verified in these separate environments.

To export security server's backup encryption public key use the following command:

    gpg --homedir /etc/xroad/gpghome --armor --output server-public-key.gpg --export <instanceIdentifier>/<memberClass>/<memberCode>/<serverCode>

where `<instanceIdentifier>/<memberClass>/<memberCode>/<serverCode>` is the security server id, for example, `AA/GOV/TS1OWNER/TS1`. 

Resulting file (`server-public-key.gpg`) should then be exported from security server and imported to GPG keystore used for backup archive consistency verification.